api/src/main/java/io/minio/CopyConditions.java

```diff
@@ -89,6 +89,20 @@ public void setMatchETagNone(String etag) throws InvalidArgumentException {
   }
 
   /**
+   * Set metadata directive to specify whether the metadata
+   * is copied from the source object or replaced with metadata
+   * provided in the request.
+   */
+  public void setMetadataDirective(MetadataDirective value) {
+    if (value == null) {
```

Comment:
Two kinds of implementation:
1. the current style.  user is required to pass correct enum even though copy is default.
2. another style is `setReplaceMetadataDirective()`.  This will add the header with `REPLACE` value. No enum necessary.

My choice is less code.  But its your call.